controllers/openstackdataplanenodeset_controller.go

```diff
@@ -192,6 +198,21 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 		instance.Status.Conditions.MarkFalse(dataplanev1.SetupReadyCondition, condition.RequestedReason, condition.SeverityInfo, condition.ReadyInitMessage)
 	}
 
+	// If we are reconciling an existing resource, compare last-applied-configuration with current spec to
+	// determine if any changes are requied. Set the status accordingly.
+	LastAppliedAnnotation := instance.GetAnnotations()["kubectl.kubernetes.io/last-applied-configuration"]
+	if LastAppliedAnnotation != "" {
+		var lastAppliedConfiguration LastAppliedConfiguration
+		if err := json.Unmarshal([]byte(LastAppliedAnnotation), &lastAppliedConfiguration); err != nil {
+			return ctrl.Result{}, err
+		}
+		if !reflect.DeepEqual(lastAppliedConfiguration.Spec.Nodes, instance.Spec.Nodes) || !reflect.DeepEqual(lastAppliedConfiguration.Spec.NodeTemplate, instance.Spec.NodeTemplate) {
```

Comment:
reflect is pretty expensive. It might be more efficient to just hash the values for each of these sections together and write that hash value instead of just a bool. Then just compare hashes each time we reconcile.

Happy to take feedback on this and implement whichever method we feel is more appropriate. The benefit of this way is that the user gets a clear indication that the configuration has been changed, whereas hash values might not be as intuitive.

Reply:
Alternative proposal is here:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/491